src/gmail.js

```diff
@@ -3442,17 +3442,28 @@ var Gmail = function(localJQuery) {
         }
 
         if (typeof element === "string") {
-            this.id = element;
-            element = $("div.adn[data-legacy-message-id='" + this.id + "']");
+            if (element.substring(0,3) === 'msg') {
+                this.id = element;
+                element = $("div.adn[data-message-id='#" + this.id + "']");
+            } else {
+                this.legacy_email_id = element;
+                element = $("div.adn[data-legacy-message-id='" + this.legacy_email_id + "']");
+            }   
         } else {
```

Comment:
Consider using the existing functions in the `api.helper.get`-namespace.

Both `api.helper.get.legacy_email_id()` and `api.helper.get.new_email_id()` cover what you are trying to do here, in what may be a more reusable/simple way. These are designed for maximum ergonomics and have pretty good test-coverage, so you should be able to rely on them acting properly (and consistently with the rest of the code-base).

For now I advice using `api.helper.get.legacy_email_id()` to maximize backwards compatibility.

Sounds good?